app/assets/javascripts/image-upload-markdown.js

```diff
@@ -0,0 +1,60 @@
+// This is a modified version of https://github.com/stuartnelson3/blog/blob/master/public/js/script.js
+
+
+$(document).on('drop', '.js-markdown-image-upload', function(e) {
+
+  e.preventDefault();
+  var dt = e.originalEvent.dataTransfer;
+  for (var i = 0; i < dt.files.length; i++) {
+    var file = dt.files[i];
+    var textarea = e.currentTarget;
+    handleFileUpload(textarea, file);
+  }
+
+  function addImageMarkdownPlaceholder(e, filename) {
+    var startPos = e.selectionStart;
+    var endPos = e.selectionEnd;
+    e.value = e.value.substring(0, startPos)
+      + "![Uploading " + filename + "...]()\r\n"
+      + e.value.substring(endPos, e.value.length);
+  }
+
+  function replaceImageMarkdownPlaceholder(textarea, value, filename){
+    var re = new RegExp("!\\[Uploading " + filename + "\\.\\.\\.\\]\\(\\)");
+    textarea.value = textarea.value.replace(re, function($1) {
+      return $1.replace("![Uploading " + filename + "...]()", value);
+    });
+  }
+
+  function handleFileUpload(textarea, file) {
+    addImageMarkdownPlaceholder(textarea, file.name);
+
+    var fd = new FormData();
+    fd.append("format", "js");
+    fd.append("file", file);
+
+    // CSRF token
+    fd.append("authenticity_token", $('meta[name="csrf-token"]').attr("content"));
+
+    var request = new XMLHttpRequest();
+    request.open("POST", "/uploads");
+
+    request.onloadend = function(filename) {
+      // Form a closure around the filename.
+      return function(e) {
+        if(e.currentTarget.status == 201){
+          var filePath = e.currentTarget.response;
+          // Update the text with the correct path for the image.
+          replaceImageMarkdownPlaceholder(textarea, "!["+filename+"]("+filePath+")", filename);
+        } else {
+          // Remove the placeholder markup and show alert
+          replaceImageMarkdownPlaceholder(textarea, "", filename);
+          alert("File upload failure\n\n" + e.currentTarget.response);
+        }
+        $(textarea).keyup();
```

Comment:
We do `e.preventDefault()` on top, I think it's to trigger the event to any potential JS code that might rely on it (this was copied over from the source, URL on top of the file).